src/printer.js

```diff
@@ -133,31 +180,74 @@ function printMemberChain(path, options, print) {
   //    PropertyLookup d, Call]
   const printedNodes = [];
 
-  function traverse(path) {
+  // Here we try to retain one typed empty line after each call expression or
+  // the first group whether it is in parentheses or not
+  function shouldInsertEmptyLineAfter(node) {
+    const { originalText } = options;
+    const nextCharIndex = getNextNonSpaceNonCommentCharacterIndex(
+      originalText,
+      node,
+      options
+    );
+    const nextChar = originalText.charAt(nextCharIndex);
+
+    // if it is cut off by a parenthesis, we only account for one typed empty
+    // line after that parenthesis
+    if (nextChar === ")") {
+      return isNextLineEmptyAfterIndex(
+        originalText,
+        nextCharIndex + 1,
+        options
+      );
+    }
+
+    return isNextLineEmpty(originalText, node, options);
+  }
+
+  function rec(path) {
     const node = path.getValue();
-    if (node.kind === "call") {
+    if (
+      node.kind === "call" &&
+      (isMemberish(node.what) || node.what.type === "call")
+    ) {
       printedNodes.unshift({
         node,
-        printed: concat([printArgumentsList(path, options, print)])
+        printed: concat([
+          printArgumentsList(path, options, print),
+          shouldInsertEmptyLineAfter(node) ? hardline : ""
+        ])
       });
-      path.call(what => traverse(what), "what");
-    } else if (node.kind === "propertylookup") {
+      path.call(what => rec(what), "what");
+    } else if (isMemberish(node)) {
+      let printedMemberish = null;
+      if (node.kind === "propertylookup") {
+        printedMemberish = printPropertyLookup(path, options, print);
+      } else if (node.kind === "staticlookup") {
+        printedMemberish = printStaticLookup(path, options, print);
+      } else {
+        printedMemberish = printOffsetLookup(path, options, print);
+      }
       printedNodes.unshift({
         node,
-        printed: concat([
-          "->",
-          wrapPropertyLookup(node, path.call(print, "offset"))
-        ])
+        printed: printedMemberish
       });
-      path.call(what => traverse(what), "what");
+      path.call(what => rec(what), "what");
     } else {
       printedNodes.unshift({
         node,
         printed: path.call(print)
       });
     }
   }
-  traverse(path);
+  // Note: the comments of the root node have already been printed, so we
```

Comment:
this comment doesn't really make sense - can you clarify?

Reply:
@mgrip right now we don't print comments on 100% as prettier, it is for future, i remove this comment from this PR and add this in future again with example